### PR TITLE
[Feature] Support mim

### PR DESCRIPTION
**Pull request by mzr1996**
1. Add mim related files to `.gitignore`
2. Generate mim related files when installing
3. Add mim introduction in docs

Refers to https://github.com/open-mmlab/mmpose/pull/706

### CODE REVIEW

**Comment by mzr1996** [comment]
@zhouzaida
mmcls mim support

File: MANIFEST.in [hunk 1]
```diff
@@ -1,3 +1,4 @@
-include mmcls/model-index.yml
-recursive-include mmcls/configs *.py *.yml
-recursive-include mmcls/tools *.sh *.py
+include mmcls/.mim/model-index.yml
+recursive-include mmcls/.mim/configs *.py *.yml
+recursive-include mmcls/.mim/tools *.py *.sh
+recursive-include mmcls/.mim/demo *.py
```
Comment (zhouzaida) [inline MANIFEST.in:4]: This line should be removed

File: setup.py [hunk 2]
```diff
@@ -90,29 +95,79 @@ def gen_packages_items():
...
+        # installed by `pip install -e .`
+        mode = 'symlink'
+    elif 'sdist' in sys.argv or 'bdist_wheel' in sys.argv:
+        # installed by `pip install .`
+        # or create source distribution by `python setup.py sdist`
+        mode = 'copy'
+    else:
+        return
+
+    filenames = ['tools', 'configs', 'demo', 'model-index.yml']
```
Comment (zhouzaida) [inline setup.py:117]:
```suggestion
    filenames = ['tools', 'configs', 'model-index.yml']
```

**Review by Ezra-Yu — APPROVED** [review]
LGTM

**Review by zhouzaida — APPROVED** [review]
LGTM

**Pull request merged**